src/lualib/StringValueOf.ts

```diff
@@ -0,0 +1,16 @@
+import { __TS__String } from "./String";
+import { String } from "./StringClass";
+
+export function __TS__StringValueOf(this: unknown): string | undefined {
```

Comment:
I also don't see the added value of adding these valueOf functions.

Reply:
I added those to assist with the test cases, mainly for the Number class as it didn't have a way of getting the raw number.